CODEOWNERS

```diff
@@ -1,5 +1,112 @@
 # Code in the `rasa.shared` package is potentially re-used by downstream dependencies
 # such as Rasa X. Hence, changes within this package require double checking.
-/rasa/shared/ @RasaHQ/enable-squad
 /docs/docs/prototype-an-assistant.mdx @ricwo @alwx
 /.github/workflows/ @RasaHQ/infrastructure
+/rasa/cli/ @RasaHQ/enable-squad
+/rasa/core/actions/ @RasaHQ/enable-squad
+/rasa/core/brokers/ @RasaHQ/enable-squad
+/rasa/core/channels/ @RasaHQ/enable-squad
+/rasa/core/featurizers/
+/rasa/core/nlg/ @RasaHQ/enable-squad
+/rasa/core/policies/ensemble.py @RasaHQ/enable-squad
+/rasa/core/policies/policy.py @RasaHQ/enable-squad
+/rasa/core/policies/registry.py @RasaHQ/enable-squad
+/rasa/core/policies/__init__.py @RasaHQ/enable-squad
+/rasa/core/policies/fallback.py @RasaHQ/research
+/rasa/core/policies/form_policy.py @RasaHQ/research
+/rasa/core/policies/mapping_policy.py @RasaHQ/research
+/rasa/core/policies/memoization.py @RasaHQ/research
+/rasa/core/policies/rule_policy.py @RasaHQ/research
+/rasa/core/policies/sklearn_policy.py @RasaHQ/research
+/rasa/core/policies/ted_policy.py @RasaHQ/research
+/rasa/core/policies/two_stage_fallback.py @RasaHQ/research
+/rasa/core/training/ @RasaHQ/enable-squad
+/rasa/core/__init__.py @RasaHQ/enable-squad
+/rasa/core/agent.py @RasaHQ/enable-squad
+/rasa/core/config.py @RasaHQ/enable-squad
+/rasa/core/exceptions.py @RasaHQ/enable-squad
+/rasa/core/exporter.py @RasaHQ/enable-squad
+/rasa/core/interpreter.py @RasaHQ/enable-squad
+/rasa/core/jobs.py @RasaHQ/enable-squad
+/rasa/core/lock.py @RasaHQ/enable-squad
+/rasa/core/lock_store.py @RasaHQ/enable-squad
+/rasa/core/processor.py @RasaHQ/enable-squad
+/rasa/core/registry.py @RasaHQ/enable-squad
+/rasa/core/run.py @RasaHQ/enable-squad
+/rasa/core/test.py @RasaHQ/enable-squad
+/rasa/core/tracker_store.py @RasaHQ/enable-squad
+/rasa/core/train.py @RasaHQ/enable-squad
+/rasa/core/utils.py @RasaHQ/enable-squad
+/rasa/core/visualize.py @RasaHQ/enable-squad
```

Comment:
this seems to be mostly enable-squad - what about making enable-squad default for `rasa/core/*` and then just adding the exceptions for research?

Reply:
Sure, I'll action that - how do I make sure that a certain file has no codeowners, would this work?
`! /rasa/core/constants.py @RasaHQ/enable-squad`

Reply:
oh, very good catch! Should we just try out? 🙈

Reply:
Alright, I'm thinking I should create branches for 4 distinct changes in: 
-  Enable owned file
-  file without owner
-  Research owned file
- dual Enable-Research ownership
I'll make some minimal change (add a comment this is for testing CODEOWNERS so it acts as a heads-up 😄 )

Reply:
Hmm, first test didn't work out :( I created [this test PR](https://github.com/RasaHQ/rasa/pull/8932) but without the expected result. Not sure where things went wrong, should I target `main` instead?

Reply:
I'm actually wondering if this syntax is valid: `/rasa/**.py @RasaHQ/enable-squad` - where did you find this in the docs? All I could find is the below - which actually states the opposite (that nested files won't be targeted):
```
# The `docs/*` pattern will match files like
# `docs/getting-started.md` but not further nested files like
# `docs/build-app/troubleshooting.md`.
docs/*  docs@example.com
```
Maybe the rule I should use is `rasa/ @RasaHQ/enable-squad` which will target **all** files in this directory as instructed here:
```
# In this example, @octocat owns any file in an apps directory
# anywhere in your repository.
apps/ @octocat
```